src/main/java/net/coderbot/iris/pipeline/AttributeShaderTransformer.java

```diff
@@ -0,0 +1,40 @@
+package net.coderbot.iris.pipeline;
+
+import net.coderbot.iris.gl.shader.ShaderType;
+import net.coderbot.iris.shaderpack.transform.StringTransformations;
+import net.coderbot.iris.shaderpack.transform.Transformations;
+
+public class AttributeShaderTransformer {
+	public static String patch(String source, ShaderType type) {
+		if (source.contains("iris_")) {
+			throw new IllegalStateException("Shader is attempting to exploit internal Iris code!");
+		}
+
+		StringTransformations transformations = new StringTransformations(source);
+
+		//Add entity color -> overlay color attribute support.
+		// TODO: We don't handle the geometry shader here.
+		if (type == ShaderType.VERTEX) {
```

Comment:
We should try to handle geometry shaders. If you can't figure out a way I have an idea.

Reply:
Feel free, I don't know what to put as the array number.

Reply:
Since it's the same for all vertices, you can just use the first element (`[0]`), assuming you're referring to what I think you're referring to.

Reply:
Ah, then I think I can implement this.

Reply:
How can we check for the existence of the use of entityColor without catching false positives like "entityColorOutput"?

Reply:
I don't think we need to check for its existence - we should just always pass it through in the geometry shader, and always make it available in the vertex shader, then the driver can optimize it out if needed.

Reply:
No, the issue I'm talking about is catching things with the name "entityColor" that don't actually refer to entityColor.

Reply:
Why would we need to do that?

Reply:
Does what I've put look good?